go/teams/handler.go

```diff
@@ -186,3 +186,42 @@ func assertCanAcceptKeybaseInvite(ctx context.Context, g *libkb.GlobalContext, u
 
 	return fmt.Errorf("chain keybase invite link eldest seqno %d does not match eldest seqno %d in team.sbs message", chainUV.EldestSeqno, untrustedInviteeFromGregor.EldestSeqno)
 }
+
+func HandleOpenTeamAccessRequest(ctx context.Context, g *libkb.GlobalContext, msg keybase1.TeamOpenReqMsg) (err error) {
+	ctx = libkb.WithLogTag(ctx, "CLKR")
+	defer g.CTrace(ctx, "HandleOpenTeamAccessRequest", func() error { return err })()
+
+	team, err := Load(ctx, g, keybase1.LoadTeamArg{
+		ID:          msg.TeamID,
+		ForceRepoll: true,
+	})
+	if err != nil {
+		return err
+	}
+
+	if !team.IsOpen() {
+		g.Log.CDebugf(ctx, "team %q is not an open team", team.Name)
+		return nil // Not an error - let the handler dismiss the message.
+	}
+
+	var req keybase1.TeamChangeReq
+	role := team.chain().inner.OpenTeamJoinAs
+
+	for _, tar := range msg.Tars {
+		uv := NewUserVersion(tar.Uid, tar.EldestSeqno)
+		if team.IsMember(ctx, uv) {
```

Comment:
If the team is opened for `WRITER` and `uv` is currently a `READER` would it make sense to let `uv` become a `WRITER`?

Reply:
I'd say handling access requests only applies to people who were not in the team at the time. If someone managed to sneak in (added by other admin in the meantime), we should leave them be. 

On the other hand, just changing this code to get and compare roles is easy. Something like this would be fine, right?
```
diff --git a/go/teams/handler.go b/go/teams/handler.go
index c1b8c4e..914966e 100644
--- a/go/teams/handler.go
+++ b/go/teams/handler.go
@@ -209,7 +209,12 @@ func HandleOpenTeamAccessRequest(ctx context.Context, g *libkb.GlobalContext, ms
 
 	for _, tar := range msg.Tars {
 		uv := NewUserVersion(tar.Uid, tar.EldestSeqno)
-		if team.IsMember(ctx, uv) {
+		currentRole, err := team.MemberRole(ctx, uv)
+		if err != nil {
+			return err
+		}
+
+		if currentRole >= role {
 			continue
 		}
 
```